File: cgi-bin/name.pl
#!/usr/bin/perl
use strict;
use warnings;

# Send HTTP header manually
print "Content-Type: text/html\r\n\r\n";

# Read query string or POST body
my $query = '';

# Handle GET or POST
if ($ENV{'REQUEST_METHOD'} eq 'GET') {
    $query = $ENV{'QUERY_STRING'} || '';
}
elsif ($ENV{'REQUEST_METHOD'} eq 'POST') {
    my $content_length = $ENV{'CONTENT_LENGTH'} || 0;
    read(STDIN, $query, $content_length);
}

# Parse parameters manually
my %params;
foreach my $pair (split /&/, $query) {
    my ($key, $value) = split /=/, $pair, 2;
    $key =~ tr/+/ /;
    $value =~ tr/+/ /;
    $key =~ s/%([0-9A-Fa-f]{2})/chr(hex($1))/eg;
    $value =~ s/%([0-9A-Fa-f]{2})/chr(hex($1))/eg;
    $params{$key} = $value;
}

# Get 'name' parameter
my $name = $params{'name'} || 'Guest';

# Print HTML
print <<EOF;
<html>
<head>
    <title>Hello CGI Test</title>
    <style>
        body { font-family: 'Segoe UI', Arial, sans-serif; background-color: #f5f5f5; color: #333; padding: 20px; }
        h1 { color: #3498db; }
        p { font-size: 1.2rem; }
    </style>
</head>
<body>
    <h1>Hello, $name!</h1>
    <p>Welcome to the WebServ CGI Perl Test Suite. Your name has been received and displayed successfully.</p>
    <p><a href="/" class="btn">Go back to the Test Suite</a></p>
</body>
</html>
EOF

File: cgi-bin/name.py
#!/usr/bin/env python3
import cgi
import cgitb
cgitb.enable()

print("Content-type: text/html\n")
print("<html><body>")
print("<h1>Python CGI Test</h1>")

form = cgi.FieldStorage()
if "name" in form:
    print(f"<p>Hello, {form['name'].value}!</p>")
else:
    print("<p>No name provided</p>")

print("</body></html>")
print("<style>@import url('https://fonts.cdnfonts.com/css/jetbrains-mono-2');:root{--font-family:\"JetBrains Mono\",monospace;--line-height:1.20rem;--border-thickness:2px;--text-color:#000;--text-color-alt:#666;--background-color:#fff;--background-color-alt:#eee;--font-weight-normal:500;--font-weight-medium:600;--font-weight-bold:800}body{font-family:var(--font-family);font-weight:var(--font-weight-normal);color:var(--text-color);background-color:var(--background-color);max-width:1000px;margin:0 auto;padding:2rem}.header{border-bottom:var(--border-thickness) solid var(--text-color);margin-bottom:2rem}.section{border:var(--border-thickness) solid var(--text-color);padding:1rem;margin-bottom:1rem}.method-badge{background:var(--background-color-alt);padding:0.2rem 0.5rem;font-size:0.8rem}button,.btn{font-family:var(--font-family);background:var(--text-color);color:var(--background-color);border:none;padding:0.5rem 1rem;cursor:pointer;text-decoration:none;display:inline-block;margin:0.25rem}input,select{font-family:var(--font-family);border:var(--border-thickness) solid var(--text-color);padding:0.5rem;margin:0.5rem 0;width:50%}.form-group{margin-bottom:1rem}</style>")